### DISCUSSION TITLE
No atom within a given radius.

### DISCUSSION THREAD

**Discussion by sharma1420** [project-gemmi/gemmi#241] [Q&A]
May you please help me,

I have a PDB format of crystal structure file( LiV3(TeO6)2. I want to find the no of atoms around each Li within a 4 A radius in the unit cell. I try to replicate the same as following,

st = gemmi.read_structure('/Users/rahul/Downloads/structure.pdb')
ns = gemmi.NeighborSearch(st[0],st.cell,4).populate(include_h=False)
ref_atom = st[0].sole_residue('',gemmi.SeqId('1')).sole_atom('Li')
marks = ns.find_neighbors('Li',min_dist=0.1,max_dist=4)
len(marks)

**Comment by wojdyr** [comment]
Did you get an error?
Note that in your script the first argument to `find_neighbors` is `'Li'` instead of `ref_atom`.

**Reply by sharma1420** [reply]
Yes, sir, I did it but got No such element Li Error

I attached the same screenshot,
![Screenshot 2022-11-22 at 7 43 12 PM](https://user-images.githubusercontent.com/118822279/203336201-b6d609bf-ba31-47a3-9805-d9b829caee7a.jpg)

**Reply by wojdyr** [reply]
It means that the residue 1 doesn't contain atom named `Li`.
In line 4: ref_atom is variable name, it should not be in quotes.

**Reply by sharma1420** [reply]
Thank you, sir, for the kind reply.

**Reply by sharma1420** [reply]
May you please guide me on exactly what I am doing wrong here?

I attached the error and Crystal structure file screenshot.

![crystal_srtucture](https://user-images.githubusercontent.com/118822279/203528510-01873f2e-c2cb-4d5b-8ad2-d3182ffc756e.jpg)
![error](https://user-images.githubusercontent.com/118822279/203528521-8a44e167-a95c-4a51-99e2-97814a2dff78.jpg)

**Reply by wojdyr** [reply]
In your previous screenshot you got the residue correctly (the chain name is empty in your PDB file), but the atom was not found because the name comparison is case sensitive (in the file it's `LI` not `Li`).

**Reply by sharma1420** [reply]
Oh, yes, sir, I got it.

Thank you.

**Reply by sharma1420** [reply]
sorry to bother you again,

in another search,
I got an error of Run time error: Multiple alternative atom LI.

![multiple alternative atom](https://user-images.githubusercontent.com/118822279/203747070-4ffd9710-c1ca-4e10-8a81-6ac0505b39af.jpg)

**Reply by wojdyr** [reply]
It means that you have multiple atom sites with the same name (alternative conformations).
`sole_atom('LI')` checks if there is only one such atom. You can use `find_atom()` instead.
